packages/ember-data/lib/system/record_array_manager.js

```diff
@@ -261,7 +261,7 @@ export default Ember.Object.extend({
   willDestroy: function(){
     this._super();
 
-    forEach(flatten(values(this.filteredRecordArrays.values)), destroy);
+    forEach(flatten(values(this.filteredRecordArrays._values)), destroy);
```

Comment:
i suspect it can be re-written (and should be).

At some point we will prefer a proper and full polyfil that supports the values() iterator